src/chain/electrum.rs

```diff
@@ -0,0 +1,303 @@
+// This file is Copyright its original authors, visible in version control history.
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. You may not use this file except in
+// accordance with one or both of these licenses.
+
+use crate::config::{
+	Config, BDK_CLIENT_STOP_GAP, BDK_WALLET_SYNC_TIMEOUT_SECS, FEE_RATE_CACHE_UPDATE_TIMEOUT_SECS,
+	LDK_WALLET_SYNC_TIMEOUT_SECS, TX_BROADCAST_TIMEOUT_SECS,
+};
+use crate::error::Error;
+use crate::fee_estimator::{
+	apply_post_estimation_adjustments, get_all_conf_targets, get_num_block_defaults_for_target,
+	ConfirmationTarget,
+};
+use crate::logger::{log_bytes, log_error, log_info, log_trace, LdkLogger, Logger};
+
+use lightning::chain::{Confirm, Filter, WatchedOutput};
+use lightning::util::ser::Writeable;
+use lightning_transaction_sync::ElectrumSyncClient;
+
+use bdk_chain::bdk_core::spk_client::FullScanRequest as BdkFullScanRequest;
+use bdk_chain::bdk_core::spk_client::FullScanResponse as BdkFullScanResponse;
+use bdk_chain::bdk_core::spk_client::SyncRequest as BdkSyncRequest;
+use bdk_chain::bdk_core::spk_client::SyncResponse as BdkSyncResponse;
+use bdk_wallet::KeychainKind as BdkKeyChainKind;
+
+use bdk_electrum::BdkElectrumClient;
+
+use electrum_client::{Batch, Client as ElectrumClient, ElectrumApi};
+
+use bitcoin::{FeeRate, Network, Script, Transaction, Txid};
+
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+const BDK_ELECTRUM_CLIENT_BATCH_SIZE: usize = 5;
+
+pub(crate) struct ElectrumRuntimeClient {
+	electrum_client: Arc<ElectrumClient>,
+	bdk_electrum_client: Arc<BdkElectrumClient<ElectrumClient>>,
+	tx_sync: Arc<ElectrumSyncClient<Arc<Logger>>>,
+	runtime: Arc<tokio::runtime::Runtime>,
+	config: Arc<Config>,
+	logger: Arc<Logger>,
+}
+
+impl ElectrumRuntimeClient {
+	pub(crate) fn new(
+		server_url: String, runtime: Arc<tokio::runtime::Runtime>, config: Arc<Config>,
+		logger: Arc<Logger>,
+	) -> Result<Self, Error> {
+		let electrum_client = Arc::new(ElectrumClient::new(&server_url).map_err(|e| {
+			log_error!(logger, "Failed to connect to electrum server: {}", e);
+			Error::ConnectionFailed
+		})?);
+		let electrum_client_2 = ElectrumClient::new(&server_url).map_err(|e| {
```

Comment:
Perhaps something better exists than `_2`? Or make the first one `_1`?

Would you be able to use the same `ElectrumClient`?